weave/trace_server/trace_server_interface.py

```diff
@@ -292,6 +292,20 @@ class TableCreateReq(BaseModel):
     table: TableSchemaForInsert
 
 
+class TableUpdateReq(BaseModel):
+    project_id: str
+    initial_digest: str
+    # insertions performed before popping so that it is easier for
+    # caller to know the index of insertion
+    insert_rows: typing.Optional[list[tuple[int, dict[str, typing.Any]]]] = None
+    pop_digests: typing.Optional[list[str]] = None
+    append_rows: typing.Optional[list[dict[str, typing.Any]]] = None
```

Comment:
Is this necessary? Feels like this is just a special case of `insert_rows`. is it here for convenience? we should be explicit that these are guaranteed to be in tail position, regardless of whatever inserting/deleting we are doing in this same request.